astroid/brain/brain_attrs.py

```diff
@@ -0,0 +1,40 @@
+# Licensed under the LGPL: https://www.gnu.org/licenses/old-licenses/lgpl-2.1.en.html
+# For details: https://github.com/PyCQA/astroid/blob/master/COPYING.LESSER
+"""
+Astroid hook for the attrs library
+
+Without this hook pylint reports unsupported-assignment-operation
+for atrrs classes
+"""
+
+import astroid
+from astroid import MANAGER
+
+
+def is_decorated_with_attrs(node):
+    """Return True if a decorated node has
+    an attr decorator applied."""
+    if not node.decorators:
+        return False
+    for decorator_attribute in node.decorators.nodes:
+        for decorator_def in decorator_attribute.infer():
+            if decorator_def.qname() == "attr._make.attributes":
```

Comment:
Ah, I see, @brycepg, maybe we should revert back to using `as_string()`. The problem is that we should avoid doing `infer()` in transform predicates, since this could lead to circular inferences across the board, rendering everything unusable (e.g an `infer()` in a predicate could trigger a circular inference for the same object, leading to a recursion error). Sorry for misleading you down this path!